tests/test_unique_ids.py

```diff
@@ -0,0 +1,35 @@
+from bs4 import BeautifulSoup  # type: ignore
+from typer.testing import CliRunner
+from jupyter_book_to_htmlbook.xref_processing import process_ids
+
+runner = CliRunner()
+
+
+def test_unique_ids():
+    """
+    IDs should be unique; duplicated IDs should be numbered sequentially
+    and those changes should be logged. Given a list, duplicate ids should
+    be updated (with randomized number) and then returned to the list
+    """
+    existing_ids = ["foo"]
+    start = BeautifulSoup("""<h1 id="foo">Hello</h1>""", "html.parser")
+    result, additions = process_ids(start, existing_ids)
```

Comment:
Suggestion: rename `start`, `result`, and `additions` to make it easier to read what the test is doing. Perhaps something like `start_chapter`, `result_chapter`, `additional_ids`.

Reply:
I use `result` pretty uniformly elsewhere in the codebase so I'd rather stick with that, but will update the others happily!